File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int count = 0;

void recursion(string words) {
    if (words.empty()) return;
    
    char first = words[0];
    int same = 1; int diff = 0;
    
    string temp = words.substr(1);
    
    for(int i=0;i<temp.size();i++){
        if(same==diff){
            count ++;
            recursion(temp.substr(i));
            return;
        }
        
        if (first == temp[i]) same ++;
        else diff ++;
    }
    
    count ++;
}

int solution(string s) {
    recursion(s);
    return count;
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
import sys
sys.setrecursionlimit(10000)
count = 0
def seperate(words):
    global count

    if words == "":
        return
    
    first = words[0]
    same, diff = 1, 0

    for idx, word in enumerate(words[1:]):
        if same == diff:
            count += 1
            seperate(words[idx + 1:])
            return
        
        if word == first:
            same += 1
        else:
            diff += 1
    
    count += 1
    return
    
def solution(s):
    global count
    seperate(s)
    return count